WIP: Updated adaptors to be based on 4.0.0-SNAPSHOT (Jax-rs2.0)

jadelkhoury opened the pull request:
This branch is based on the latest changes in LyoDesigner 4.0.0-SNAPSHOT ( and hence Jax-rs2.0)

I think the master branch should still be based on the released 2.4.0, but we do need this branch as long as we have not released 4.0.0 - at the least.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/oslc/lyo-adaptor-sample-modelling/24)
<!-- Reviewable:end -->

jadelkhoury committed:
Updated adaptors to be based on 4.0.0-SNAPSHOT (Jax-rs2.0)

jadelkhoury commented:
@berezovskyi @jamsden 
Can you please help me with the review of the pom.xml files at the least?

I would like the pom.xml files to be as neat and minimal as possible, since I can then use them as basis to update our instructions on Wiki on how to create servers/clients in general.
Notes: 
1. There is a new dependencyManagement part, which might be a hack to make Swagger 1.5 work together with Jersey 2.27.) Is there a better way to do it?
2. I could not get the code to work on the latest Jersey 2.28. Only 2.27
3. Any thing else i missed?

The 2 adaptors work fine. THey also include client code.

berezovskyi commented on adaptor-rm-webapp/pom.xml:
```diff
@@ -91,6 +101,15 @@
+			<dependency>
+				<groupId>org.glassfish.jersey.containers</groupId>
+				<artifactId>jersey-container-servlet-core</artifactId>
+				<version>${version.jersey}</version>
```
what's this?

jadelkhoury replied:
It was an attempt to reconsile the Jersey dependency version requested by Swagger 1.5 (Jersey 2.25.1), with the latest one we want to use (2.27).
Now changing Jersey version to be 2.25.1.

berezovskyi commented:
looks good overall, so I have approved. but please see the comments inside.

jadelkhoury committed:
Update Jersey version to match that supported by Swagger 1.5.

jadelkhoury commented:
@berezovskyi Please have a look at my new patch to see if it is what you had in mind.

berezovskyi commented on adaptor-rm-webapp/pom.xml:
```diff
@@ -88,11 +90,6 @@
 			<artifactId>jersey-container-servlet</artifactId>
 			<version>${version.jersey}</version>
 		</dependency>
-		<dependency>
```
wait, why did we remove the HK2 DI dep?

jadelkhoury replied:
The earliest version of jersey-hk2 is 2.26. Code runs fine without it if we use 2.25.
But now I realize we have a bigger problem. We will need CQs if we go for Jersey 2.25, since Jersey was migrated to Eclipse in 2.27.
And that also made me think that we did NOT file a CQ for Swagger.

jadelkhoury replied:
so do I have your blessing on this patch?

berezovskyi replied:
yes, please go ahead

berezovskyi commented:
None of those dependencies are shipped with Lyo releases, no need to CQ them.

--
/Andrew
(from phone)

28 mars 2019 kl. 07:51 skrev Jad El-khoury <notifications@github.com<mailto:notifications@github.com>>:


@jadelkhoury commented on this pull request.

________________________________

In adaptor-rm-webapp/pom.xml<https://github.com/OSLC/lyo-adaptor-sample-modelling/pull/24#discussion_r269874273>:

> @@ -88,11 +90,6 @@
                        <artifactId>jersey-container-servlet</artifactId>
                        <version>${version.jersey}</version>
                </dependency>
-               <dependency>


The earliest version of jersey-hk2 is 2.26. Code runs fine without it if we use 2.25.
But now I realize we have a bigger problem. We will need CQs if we go for Jersey 2.25, since Jersey was migrated to Eclipse in 2.27.
And that also made me think that we did NOT file a CQ for Swagger.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/OSLC/lyo-adaptor-sample-modelling/pull/24#discussion_r269874273>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AAD83mvPZevmtPF2dCDoWKToWgsQ_fnrks5vbGZigaJpZM4b5wy9>.

jadelkhoury commented:
We will not merge this branch into master until 4.0.0 is released.
Master is based on 2.4.0.

berezovskyi commented:
Sounds good to me.

--
–Andrew.

2019-03-28  kl. 14:28 skrev Jad El-khoury <notifications@github.com<mailto:notifications@github.com>>:


We will not merge this branch into master until 4.0.0 is released.
Master is based on 2.4.0.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/OSLC/lyo-adaptor-sample-modelling/pull/24#issuecomment-477594591>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AAD83uC5Wn-GfjH_Hwk9FP_34sgznP2Rks5vbMN7gaJpZM4b5wy9>.

jadelkhoury committed:
update of the jsp pages to no longer rely on the removed toHtml() (#29)

methods.

berezovskyi committed:
Update logging

Signed-off-by: Andrew Berezovskyi <andriib@kth.se>

berezovskyi commented:
client needs to be updated.

berezovskyi committed:
Fix client imports

berezovskyi commented:
This branch has diverged from master and has merge conflicts. I have resolved most of them but there is a conflict in the AIRD file, which is super tricky:

![image](https://user-images.githubusercontent.com/64734/56851209-a62cf900-690c-11e9-912d-30a14b89c40b.png)

berezovskyi commented:
I have merged the AIRD from master in this case.

berezovskyi commented:
Just ignore it, I was touching the wrong branch.

berezovskyi committed:
Merge branch 'master' into 4.0.0-snapshot

berezovskyi committed:
Update Jetty

Signed-off-by: Andrew Berezovskyi <andriib@kth.se>

berezovskyi committed:
Use new OSLC4J Client

Signed-off-by: Andrew Berezovskyi <andriib@kth.se>

berezovskyi commented:
@jadelkhoury I have switched the code to use OSLC4J Client so that the build stops failing but this needs to be done in LyoD: https://github.com/OSLC/lyo-adaptor-sample-modelling/pull/24/commits/dcb169a4edd8e98a7571c416cc40e89bd1df84a6

berezovskyi committed:
Merge branch 'master' into 4.0.0-snapshot

jadelkhoury commented:
I am concerned about all changes done on this branch since https://github.com/OSLC/lyo-adaptor-sample-modelling/pull/24/commits/a4011721f26996275881d2642a61d981adf504b8

1. Changes to generated code should be avoided, since the main point of this project is to demonstrated generated code. We need to avoid manual changes (such as those done on the client).
2. Changes to the non-generated parts of the code should be captured in our instructions http://oslc.github.io/developing-oslc-applications/eclipse_lyo/setup-an-oslc-provider-consumer-application.html 
3. I'd be very very careful when merging AIRD files. (that is avoid.) THere is no point merging from master either. Master is meant to be for code modelled/generated from 2.4.0. THis branch is from 4.0.0. Naturally, they are to diverse.

jadelkhoury committed:
Update based on latest LyoDesigner 4.0.0-snapshot

berezovskyi committed:
Use latest Jetty server

Signed-off-by: Andrew Berezovskyi <andriib@kth.se>

berezovskyi committed:
Update

Signed-off-by: Andrew Berezovskyi <andriib@kth.se>

jadelkhoury committed:
closes #35 - Testing adaptor uses old client

Generating working code based on latest LyoDesigner 4.0-SNAPSHOT.
Reintroducing package imports that were removed in previous commits.

berezovskyi commented:
Do you want to drag this branch all the way till the 4.0 is released? Or shall we merge it soon?

jadelkhoury commented:
> Do you want to drag this branch all the way till the 4.0 is released? Or shall we merge it soon?

My thought is that we wait until 4.0 is released.

jadelkhoury committed:
Introducing Creation Dialog and complete CRUD functions on the rm-adaptor. (#45)

* Introducing Creation Dialog and complete CRUD functions on the
rm-adaptor.
Also, modifying the way the rm-adpator resources are randomly generated.

* Update of jsp pages.

jadelkhoury committed:
reverting a an earlier version of Sirius to make sure model can be
opened by older versions of Lyo.

jadelkhoury committed:
Generating code based on latest LyoDesigner templates.

jadelkhoury committed:
Updating adaptors to latest LyoGen

berezovskyi commented:
Do you need any help resolving merge conflicts or with anything else in this PR?

jadelkhoury commented:
> Do you need any help resolving merge conflicts or with anything else in this PR?

Most likely. my thought is that we move this into master, once we release LyoDesigner 4.0.0, which I hope we can do soon.
So first step is to release 4.0.0.

jadelkhoury committed:
Updating based on latest edge version of Lyo Designer.

jadelkhoury committed:
Adding a sample client-only adaptor.

jadelkhoury committed:
removing unecessary "throws exceptions" that are not actually called.

jadelkhoury committed:
Update to latest version of Lyo Designer generation.

jadelkhoury committed:
An update of both adaptors to be based on latest lyoDesigner.
Plus, now using WebServices for the CRUD methods.

jadelkhoury committed:
TRS support for adaptor-rm

berezovskyi commented:
I suggest to keep this as a sample JAX-RS 1.1 implementation instead of merging this and then deprecating in favour of OSLC 2020 refimpl

jadelkhoury commented:
@berezovskyi Your last comment doesn't make sense to me.
Please, let us discuss this repo before we make any chanfges to it.

berezovskyi commented:
@jadelkhoury I based it on what you wrote actually in: https://github.com/oslc-op/refimpl/issues/21 If we go that route, let's keep this repo as a valid 2.x Lyo sample code and follow your plan to make refimpl repo the Lyo 4.0 / JAX-RS 2.0 sample code.

jadelkhoury commented:
It will take some time before we do https://github.com/oslc-op/refimpl/issues/21. Until then, this repo is what I need to have.

If you want to do something on this repo, here's what I suggest (so we avoid all confusing once and for all):

We have two branches 2.4.0 & 4.0.0-SNAPSHOT. No master branch.
that is, either rename "master" into 2.4.0, or create a new branch from master to 2.4.0, and then kill the master branch.

berezovskyi commented:
Ok, my plan is not to touch master (much) especially given that no 2.x lyo releases are planned to come.

I can rename master into main-2.x and this branch into main-4.x (maybe there will be updates to this repo when 4.1 comes out?)

We can then close this PR as there is no reason to merge it anywhere and I can make main-4.x to be the main branch on Github.

jadelkhoury commented:
Yes to all in last proposal. thanks @berezovskyi

berezovskyi commented:
@jadelkhoury done. please delete outdated branches from https://github.com/OSLC/lyo-adaptor-sample-modelling/branches/stale if possible
